src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentState.cs

```diff
@@ -107,6 +114,12 @@ protected static ValueSource<TextAndVersion> CreateRecoverableText(TextLoader lo
             }
             catch (IOException e)
             {
+                if (++retries <= MaxRetries)
```

Comment:
@pharring it is a common scenario that tools such as designers rewriting documents may indeed cause them to appear deleted before rewritten.  In such a case, file watchers can trigger a reload of the document while it is missing.

Also, the max wait is only 1 sec.